fn_test.go

```diff
@@ -0,0 +1,616 @@
+package secp256k1_test
+
+import (
+	"bytes"
+	"crypto/rand"
+	"math/big"
+	mrand "math/rand"
+
+	"github.com/renproject/secp256k1/testutil"
```

Comment:
Consider renaming this module `secp256k1util` as it is more consistent with the convention we are moving to in other repos. The reason begin, you often need to import multiple utils when writing tests/etc. and having them all named `testutil` is cumbersome to rename in a way that is always easily identifiable when just reading through code.